monkey/tests/unit_tests/agent_plugins/exploiters/wmi/test_wmi_command_builder.py

```diff
@@ -0,0 +1,104 @@
+from pathlib import PureWindowsPath
+from unittest.mock import MagicMock
+
+import pytest
+from agent_plugins.exploiters.wmi.src.wmi_command_builder import build_wmi_command
+
+from common import OperatingSystem
+from infection_monkey.exploit import IAgentOTPProvider
+from infection_monkey.model import DROPPER_ARG, MONKEY_ARG
+from infection_monkey.utils.ids import get_agent_id
+
+DROPPER_EXE_PATH = PureWindowsPath("C:\\dropper.exe")
+AGENT_EXE_PATH = PureWindowsPath("C:\\agent.exe")
+OTP = "123456"
+AGENT_ID = get_agent_id()
+
+
+@pytest.fixture
+def otp_provider() -> IAgentOTPProvider:
+    provider = MagicMock(spec=IAgentOTPProvider)
+    provider.get_otp.return_value = OTP
+    return provider
+
+
+def test_exception_raised_for_linux(otp_provider: IAgentOTPProvider):
+    with pytest.raises(Exception):
+        build_wmi_command(
+            AGENT_ID,
+            ["127.0.0.1"],
+            2,
+            OperatingSystem.LINUX,
+            DROPPER_EXE_PATH,
+            AGENT_EXE_PATH,
+            otp_provider,
+        )
```

Comment:
We can put this in a fixture and use it in every test? Also they seem like they can be just one or two tests.